core/controllers/base_test.py

```diff
@@ -107,9 +120,55 @@ def test_redirect_in_both_logged_in_and_logged_out_states(self):
         self.login('user@example.com')
         response = self.testapp.get('/')
         self.assertEqual(response.status_int, 302)
+
+        # Homepage is either dashboard or library based on user's learner
+        # status, this homepage redirect rule behavior is seperately tested
+        atHome = ('dashboard' in response.headers['location'] or
+                  'library' in response.headers['location'])
+        self.assertTrue(atHome, msg="user must be at library or dashboard")
+        self.logout()
+
+    def test_root_redirect_rules_for_learners(self):
+        # Test learner behavior. Login as learner
```

Comment:
I don't think you need this comment. It's pretty clear from both the method name and the test name.